Mathlib/Computability/TuringDegree.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2025 Tanner Duve. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tanner Duve, Elan Roth
+-/
+import Mathlib.Computability.Partrec
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Oracle Computability and Turing Degrees
+
+This file defines a model of oracle computability, introduces Turing reducibility and equivalence,
+proves that Turing equivalence is an equivalence relation, and defines Turing degrees as the
+quotient under this relation.
+
+## Main Definitions
+
+- `RecursiveIn O f`:
+  An inductive definition representing that a partial function `f` is partial recursive given access
+  to a set of oracles O.
+- `TuringReducible`: A relation defining Turing reducibility between partial functions.
+- `TuringEquivalent`: An equivalence relation defining Turing equivalence between partial functions.
+- `TuringDegree`: The type of Turing degrees, defined as the quotient of partial functions under
+  `TuringEquivalent`.
+
+## Notation
+
+- `f ≤ᵀ g` : `f` is Turing reducible to `g`.
+- `f ≡ᵀ g` : `f` is Turing equivalent to `g`.
+
+## Implementation Notes
+
+The type of partial functions recursive in an oracle `g` is the smallest type containing
+the constant zero, the successor, projections, the oracle `g`, and is closed under
+pairing, composition, primitive recursion, and μ-recursion.
+
+## References
+
+* [Carneiro2018] Carneiro, Mario.
+  *Formalizing Computability Theory via Partial Recursive Functions*.
+  arXiv preprint arXiv:1810.08380, 2018.
```

Comment:
I think this is already in mathlib's bibliography, so you don't need a full ref here. If you do though, maybe it's better to give the published version: https://drops.dagstuhl.de/entities/document/10.4230/LIPIcs.ITP.2019.12